dataforest/core/PlotWidget.py

```diff
@@ -0,0 +1,83 @@
+from copy import deepcopy
+from typing import TYPE_CHECKING, Dict, Any
+
+from IPython.display import Image
+import ipywidgets as widgets
+from matplotlib.figure import Figure
+
+from dataforest.structures.cache.PlotCache import PlotCache
+
+if TYPE_CHECKING:
+    from dataforest.core.DataBranch import DataBranch
+
+
+class PlotWidget:
+    def __init__(self, tree, plot_key, use_saved=True, **plot_kwargs):
+        self._tree = tree
+        self._branch_spec = deepcopy(tree.tree_spec.branch_specs[0])
+        self._plot_key = plot_key
+        self._use_saved = use_saved
+        self._plot_kwargs = plot_kwargs
+        self._sweeps = self._tree.tree_spec.sweep_dict
+        self._plot_cache = dict()
+        # self.ax = plt.gca()
+
+    def control(self):
+        process = self._tree.current_process
+        precursors = self._tree.tree_spec.get_precursors_lookup(incl_current=True)[process]
+        sweeps = set.union(*[self._sweeps[precursor] for precursor in precursors])
+        param_sweeps = {":".join(swp[:3][::-1]): list(swp[3]) for swp in sweeps}
+        _kwargs = {**param_sweeps, **self._plot_kwargs}
+
+        @widgets.interact(**_kwargs)
```

Comment:
Nice! So this will work only in Jupyter notebooks?

Reply:
Yeah -- it would be nice to make a fancy universal JS or Flask app at some point, but this is it for now